src/components/autocomplete.tsx

```diff
@@ -0,0 +1,112 @@
+import debounce from 'lodash.debounce';
+import { useCallback, useState, useMemo } from 'react';
+import { Person } from '../types/Person';
+import cn from 'classnames';
+
+interface Props {
+  peopleFromServer: Person[];
+  onSelected: (person: Person) => void;
+  onInputChange: () => void;
+  debounceDelay?: number;
+}
+
+export const CustomAutocomplete: React.FC<Props> = ({
+  peopleFromServer,
+  onSelected,
+  onInputChange,
+  debounceDelay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [dropDown, setDropDown] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useMemo(
+    () =>
+      debounce((value: string) => {
+        setAppliedQuery(value);
+        setDropDown(true);
+      }, debounceDelay),
+    [debounceDelay],
+  );
+
+  const handleQueryChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const value = event.target.value;
+
+      setQuery(value);
+      applyQuery(value);
+      setDropDown(false);
+      onInputChange();
+    },
+    [applyQuery, onInputChange],
+  );
+
+  const handleSelectPerson = useCallback(
+    (person: Person) => {
+      setQuery(person.name);
+      setDropDown(false);
+      onSelected(person);
+    },
+    [onSelected],
+  );
+
+  const filteredPeople = useMemo(() => {
+    if (appliedQuery.trim() === '') {
+      return peopleFromServer;
+    }
+
+    return peopleFromServer.filter(person => {
+      const normalName = person.name.trim().toLowerCase();
+      const normalQuery = appliedQuery.trim().toLowerCase();
+
+      return normalName.includes(normalQuery);
+    });
+  }, [appliedQuery, peopleFromServer]);
+
+  return (
+    <div className={cn('dropdown', { 'is-active': dropDown })}>
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          data-cy="search-input"
+          value={query}
+          onChange={handleQueryChange}
+          onFocus={() => setDropDown(query.trim() === '')}
+        />
+      </div>
+
+      {dropDown && (
+        <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
+          <div
+            className={filteredPeople.length !== 0 ? 'dropdown-content' : ''}
```

Comment:
```suggestion
            className={!!filteredPeople.length ? 'dropdown-content' : ''}
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**